src/DotVVM.Framework/ResourceManagement/LocalResourceUrlManager.cs

```diff
@@ -52,16 +79,28 @@ protected virtual string DecodeResourceName(string name)
         public ILocalResourceLocation FindResource(string url, IDotvvmRequestContext context, out string mimeType)
         {
             mimeType = null;
-            if (DotvvmRoutingMiddleware.FindMatchingRoute(new[] { resourceRoute }, context, out var parameters) == null) return null;
-            var name = DecodeResourceName((string)parameters["name"]);
-            var hash = (string)parameters["hash"];
+            if (DotvvmRoutingMiddleware.FindMatchingRoute(new[] { resourceRoute }, context, out var parameters) == null)
+            {
+                return null;
+            }
+
+            var name = DecodeResourceName((string)parameters[NameParameterName]);
+            var hash = (string)parameters[HashParameterName];
+            int? locationIndex = null;
+            if (context.Query.TryGetValue(LocationQueryName, out var locationString))
+            {
+                locationIndex = (int)ReflectionUtils.ConvertValue(locationString, typeof(int));
```

Comment:
Why not `int.Parse`? This `ConvertValue` value function is not designed to take input from the network.

Reply:
I followed the logic handling query parameters until I arrived at `ParameterBindingAttribute`. `ConverValue` is used there and I wanted to be consistent.